pkg/outputs/kafka_output/kafka_output.go

```diff
@@ -398,32 +523,39 @@ func (k *kafkaOutput) createConfig() (*sarama.Config, error) {
 				return &XDGSCRAMClient{HashGeneratorFcn: SHA512}
 			}
 		case sarama.SASLTypeOAuth:
-			cfg.Net.SASL.TokenProvider = oauthbearer.NewTokenProvider(cfg.Net.SASL.User, cfg.Net.SASL.Password, k.Cfg.SASL.TokenURL)
+			cfg.Net.SASL.TokenProvider = oauthbearer.NewTokenProvider(cfg.Net.SASL.User, cfg.Net.SASL.Password, k.cfg.SASL.TokenURL)
 		}
 	}
 	// SSL or SASL_SSL
-	if k.Cfg.TLS != nil {
+	if k.cfg.TLS != nil {
 		var err error
 		cfg.Net.TLS.Enable = true
 		cfg.Net.TLS.Config, err = utils.NewTLSConfig(
-			k.Cfg.TLS.CaFile,
-			k.Cfg.TLS.CertFile,
-			k.Cfg.TLS.KeyFile,
+			k.cfg.TLS.CaFile,
+			k.cfg.TLS.CertFile,
+			k.cfg.TLS.KeyFile,
 			"",
-			k.Cfg.TLS.SkipVerify,
+			k.cfg.TLS.SkipVerify,
 			false)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	cfg.Producer.Retry.Max = k.Cfg.MaxRetry
-	cfg.Producer.RequiredAcks = sarama.WaitForAll
+	cfg.Producer.Retry.Max = k.cfg.MaxRetry
 	cfg.Producer.Return.Successes = true
-	cfg.Producer.Timeout = k.Cfg.Timeout
+	cfg.Producer.Timeout = k.cfg.Timeout
+	switch k.cfg.RequiredAcks {
+	case requiredAcksNoResponse:
```

Comment:
Should this case set the sarama config to `NoResponse`? It looks like `WaitForLocal` is the sarama default.

Reply:
Yes I wanted to follow the sarama default in this case.